File: 0700-search-in-a-binary-search-tree/0700-search-in-a-binary-search-tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public TreeNode searchBST(TreeNode root, int val) {
        // Base Case:
        if(root == null) 
            return null;
        
        // Checking if the root node is the target Node:
        if(root.val == val) return root;
        
        // Search in the Left Sub-Tree:
        TreeNode lTree = searchBST(root.left, val);
        if(lTree != null && lTree.val == val) return lTree;
        
        // Searching in the Right Sub-Tree:
        TreeNode rTree = searchBST(root.right, val);
        if(rTree != null && rTree.val == val) return rTree;
        
        // Other wise:
        return null;
    }
}

File: 0700-search-in-a-binary-search-tree/0700-search-in-a-binary-search-tree.js
/**
 * Definition for a binary tree node.
 * function TreeNode(val, left, right) {
 *     this.val = (val===undefined ? 0 : val)
 *     this.left = (left===undefined ? null : left)
 *     this.right = (right===undefined ? null : right)
 * }
 */
/**
 * @param {TreeNode} root
 * @param {number} val
 * @return {TreeNode}
 */
var searchBST = function(root, val) {
    // Base Case:
    if(!root || root.val == val) 
        return root;
    
    // Recursive Call:
    // Searching in the Left Tree:
    let lTree = searchBST(root.left, val);
    // Checking the Ans:
    if(lTree && lTree.val === val) return lTree;
    
    // Searching in the Right Tree:
    let rTree = searchBST(root.right, val);
    // Checking the Ans:
    if(rTree && rTree.val === val) return rTree;

    // Else:
    return null;
};